### PR TITLE
Improve Travis test coverage

### PR DESCRIPTION
**Pull request by justinseanmartin**
All the test issues that I'm addressing in my other PRs should be covered by increasing the automated test matrix a bit. This will show the issues those PRs are addressing as well as ensure they don't regress in the future. 

This shouldn't be merged until #777, #776, #775, #774, #773, #772, #771 are all merged, and this PR has been rebased to show that Travis will go green.

### CODE REVIEW

**Comment by justinseanmartin** [comment]
This doesn't work quite like I expected, will investigate further and play around with this.

**Comment by justinseanmartin** [comment]
Ok, just rebased with all the test fixes merged. Hopefully this should go green now.

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+    - osx_image: xcode6.4
+      env: 'SIMULATOR="name=iPhone 4s,OS=8.4"'
+    - env: 'SIMULATOR="name=iPad 2,OS=7.1"' # Default: Xcode 6.1
 
 before_install:
   - xcrun simctl list
   - brew update || brew update
 install:
-  - brew install coreutils
+  - if [ -z "$(brew ls --versions coreutils)" ] ; then brew install coreutils ; fi
```
Comment (justinseanmartin) [inline .travis.yml]: I was seeing a bunch of "ERROR: coreutils is already installed", so this seemed like a reasonable way to avoid that error.

**Comment by phatmann** [comment]
Travis build failed. Rerunning job now.

**Comment by justinseanmartin** [comment]
I just rebased with the latest test fixes, which will help.

**Comment by justinseanmartin** [comment]
Woo, green builds finally!!

**Comment by phatmann** [comment]
Woot, that is something to see! Thanks so much for all your efforts here, the whole project will benefit greatly from it.

**Comment by mikelupo** [comment]
Great work Justin. Thanks for your contributions!!

**Pull request merged**